src/library/scala/collection/View.scala

```diff
@@ -37,6 +35,12 @@ trait View[+A] extends Iterable[A] with IterableOps[A, View, View[A]] with Seria
 
   @deprecated("Views no longer know about their underlying collection type; .force always returns an IndexedSeq", "2.13.0")
   @`inline` def force: IndexedSeq[A] = toIndexedSeq
+
+  /**
+    * @return a view containing the elements of this view which only evaluates elements once,
+    *         and does so lazily.
+    */
+  def cached: View[A] = LazyList.from(this).view
```

Comment:
Why go through LazyList in particular ?

Reply:
it's lazy so it has good caching semantics - i.e. it doesn't require evaluating the entire view if you access a single element

Reply:
it's lazy so it has good caching semantics - i.e. it doesn't require evaluating the entire view if you access a single element.

also, I'm not sure what the alternative is